src/hooks/useBalance/PickBalanceProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import type { TUseBalanceOptions } from './types'
+import { useCosmosBalance } from './useCosmosBalance'
+import { useEVMBalance } from './useEVMBalance'
+import { useSolanaBalance } from './useSolanaBalance'
+
+function EVMBalanceComponent(p: any) {
+  const balance = useEVMBalance(p.options)
+  return p.children(balance)
+}
+
+function CosmosBalanceComponent(p: any) {
+  const balance = useCosmosBalance(p.options)
+  return p.children(balance)
+}
+
+function SolanaBalanceComponent(p: any) {
+  const balance = useSolanaBalance(p.options)
+  return p.children(balance)
+}
+
+const BALANCE_PROVIDER_BY_NAME = {
+  MetaMask: EVMBalanceComponent,
+  WalletConnect: EVMBalanceComponent,
+  Coinbase: EVMBalanceComponent,
+  xDefi: EVMBalanceComponent,
+  Phantom: SolanaBalanceComponent,
+  Near: null,
+  Keplr: CosmosBalanceComponent
+}
+
+function PickBalanceProvider(props: {
+  options: TUseBalanceOptions
+  children: (balance: string | null) => React.ReactNode
+}) {
+  const { options, children } = props
+  const { name } = options
+
+  if (!name) {
+    return null
+  }
+
+  const BalanceComponent = BALANCE_PROVIDER_BY_NAME[name]
```

Comment:
@nrjshka  Hi, sorry. I'm not a member, just a guest. Why do you need useMemo? I created a simple [example](https://codesandbox.io/s/password-box-antd-4-20-2-forked-dsnyp9?file=/demo.js) for researching re-rendering. I think your proposal is overthinking. BalanceComponent will be re-rendered when using useMemo or not for BALANCE_PROVIDER_BY_NAME.

Reply:
It happens because you pass options as prop to <BalanceComponent>. if you want to prevent re-render for BalanceComponent when children or options are not changed, just wrap
```
return useMemo(() => {
  if (!BalanceComponent) return null;
 
  return <BalanceComponent options={options}>{children}</BalanceComponent>;
}, [options, children])
```
but i don't think this optimization is actually needs